### ISSUE TITLE
Question: should scale_by_points be disabled for public pose-only Cambrian-P-Data?

### ISSUE THREAD

**Issue by Grady10086** [cambrian-mllm/cambrian-p#1]
Hi, thanks for releasing Cambrian-P.

I am trying to reproduce Cambrian-P training with the released public data:
- `nyu-visionx/vsi-590k`
- `nyu-visionx/Cambrian-S-3M`
- `nyu-visionx/Cambrian-P-Data`

I noticed a possible mismatch between the public pose-only annotations and the current training loader.

In `load_vipe_cambrians_scene`, when `depth_path` is missing, the loader fabricates zero depth maps and therefore `point_masks = depths_192 > 0` becomes all false. However, the returned `rec_views` still sets:

scale_by_points: `torch.tensor(True)`

Later, normalize_camera_extrinsics_and_points_batch(..., scale_by_points=True, point_masks=...) computes the average scene scale from point_masks. Since the mask is empty, avg_scale is clamped to 1e-6, and camera translations are divided by 1e-6. This appears to inflate the camera loss significantly.

In our run, the first logged step before this change had approximately:

rec_loss_camera: ~54
rec_all: ~273
After changing the loader to only scale by points when depth exists:

- "scale_by_points": torch.tensor(True)
+ "scale_by_points": torch.tensor(has_depth)
the first logged step became:

rec_loss_camera: 1.0866
rec_all: 5.4331
I wanted to ask what the intended official behavior is for the released public pose-only Cambrian-P-Data:

Should scale_by_points be disabled when depth_path is missing?
Should camera loss use a frame-valid mask independent from point_masks for pose-only supervision?
Is the released training expected to run with --use_point_masks False when only camera supervision is enabled?
Or is there an additional depth / point-mask release or preprocessing step expected by the official recipe?
A minimal defensive fix could be either:

set scale_by_points = has_depth in the ViPE-Cambrians loader, or
inside normalization, skip point-based scaling when point_masks.sum() == 0.
Could you confirm which fix matches the intended Cambrian-P training setup?

**Comment by Exiam6** [comment]
Hello! Thanks for the careful write-up and for verifying the numbers.

You are right, there were issues in the scale_by_points and use_point_masks settings for the released setup. The suggested changes are correct, and we have applied the corresponding fixes on our side.

Answering your specific questions:
1. Should scale_by_points be disabled when depth_path is missing?
Yes. The hardcoded True was a bug introduced during code cleanup. Conceptually, point-based scaling is undefined without valid depth/point supervision, so scale_by_points should be disabled when depth is unavailable. 

2. Should the camera loss use a frame-valid mask independent of point_masks?
For pose-only supervision, all pose-supervised frames are treated as valid.

3. Should training run with --use_point_masks False for camera-only / pose-only supervision?
Yes. This is required for the released pose-only Cambrian-P-Data setup. We have also updated the provided training scripts accordingly.

4. Is any additional depth / point-mask release or preprocessing step expected?
No. Cambrian-P-Data is pose-only by design. The fabricated-zero-depth branch is the intended path for the released data.

Thanks again for the precise report!
